src/main/java/com/google/sps/api/GmailAPI.java

```diff
@@ -90,20 +93,38 @@ public static Gmail getGmailService() throws IOException, GeneralSecurityExcepti
   }
 
   /**
-  * Retrieve new access token due to the fact that it has a lifetime of 1 hour
-  * through refresh token.
-  *
-  * @param refresh_token Authorized token to create access tokens.
-  */
-  private static String getAccessToken() 
+    * Retrieve secrets throgh Secrets Manager.
+    *
+    * @param secretId id of secret key
+    * @throws IOException
+    */
+  public static String getSecret(String secretId) throws IOException {
+
+    // Retrieve the secret key for given ID.
+    try (SecretManagerServiceClient client = SecretManagerServiceClient.create()) {
```

Comment:
Is there a reason there's no catch block here?

At the very least, add a TODO so you can get back to it.